modules/calendar/event.py

```diff
@@ -4,78 +4,122 @@
+		return self.__description
+
+	@property
 	def all_day(self) -> bool:
 		"""Returns whether or not the event is an all day event"""
-		return "date" in self.details["start"]
+		return self.__all_day
 
+	@property
 	def start(self) -> datetime:
 		"""Returns the start date as a datetime object"""
-		return self.__get_endpoint("start")
+		return self.__start
 
+	@property
 	def end(self) -> datetime:
 		"""Returns the end date as a datetime object"""
-		return self.__get_endpoint("end")
+		return self.__end
 
-	def start_str(self) -> str:
-		"""Returns a formatted string of the start date"""
-		return format_date(self.start(), all_day=self.all_day()) or "Today"
+	@property
+	def timezone(self) -> str:
+		"""Returns the timezone passed to the constructor or the event's timezone if not specified. If neither are present, returns None."""
+		return self.__timezone
 
-	def end_str(self, base=datetime.now()) -> str:
-		"""Returns a formatted string of the end date"""
-		end_date = self.end()
-		# use previous day if end of multi-day, all-day event
-		if self.all_day() and not self.__one_day():
-			end_date -= timedelta(days=1)
-		return format_date(end_date, all_day=self.all_day(), base=base)
+	@property
+	def __one_day(self) -> bool:
+		"""Returns whether or not the event is a one day event"""
+		return self.all_day and self.end - self.start <= timedelta(days=1)
 
-	def date_range_str(self) -> str:
+	def relative_date_range_str(self, base=datetime.now()) -> str:
 		"""Returns a formatted string of the start to end date range"""
-		start_str = self.start_str()
-		end_str = self.end_str(base=self.start())
+		start_str = self.__relative_start_str(base=base)
+		end_str = self.__relative_end_str(base=self.start)
 		# all day event
-		if self.__one_day():
+		if self.__one_day:
 			return f"{start_str} - All day"
 		# include end time if it is not the same as the start time
 		return f"{start_str} - {end_str}" if end_str else start_str
 
-	def timezone(self) -> str:
-		"""Returns the timezone passed to the constructor \
-			or the event's timezone if not specified. \
-			If neither are present, returns None."""
-		return self.tz
-
-	def location(self) -> str:
-		"""Returns the location of the event"""
-		return self.details.get('location')
+	def __relative_start_str(self, base=datetime.now()) -> str:
+		"""Returns a formatted string of the start date"""
+		return format_date(self.start, all_day=self.all_day, base=base) or "Today"
 
-	def description(self) -> str:
-		"""Returns the description of the event"""
-		return self.details.get('description')
+	def __relative_end_str(self, base=datetime.now()) -> str:
+		"""Returns a formatted string of the end date"""
+		end_date = self.end
+		# use previous day if end of multi-day, all-day event
+		if self.all_day and not self.__one_day:
+			end_date -= timedelta(days=1)
+		return format_date(end_date, all_day=self.all_day, base=base)
+
+	@classmethod
+	def from_dict(cls, details: Dict[str, str]) -> "Event":
```

Comment:
```suggestion
	def from_dict(cls, details: Dict[str, Any]) -> "Event":
```
could be nested dicts, not necessarily str